### DISCUSSION TITLE
Back Arrow on MASS entry screen

### DISCUSSION THREAD

**Discussion by FrankJaco** [music-assistant/support#936] [Feature requests and ideas]
I would like to see a back arrow on the initial Music Assistant screen (as it already appears on the various child MASS screens).
I think a common way to access MASS is via a button helper that navigates to the MASS initial screen. However if that button is pressed erroneously, the way only back is the browser back button. Having a back arrow on the initial screen would give getting into MASS a user interface consistent with the rest of MASS.  AND with the release of HA 2022.10 and the new "Subview" feature, this consistent interface will now be common throughout Home Assistant as well.

![image](https://user-images.githubusercontent.com/79318542/194721957-ddb6af27-0eb3-42fa-beb5-8221fedaa73c.png)

**Comment by corvy** [comment]
I agree to this. Also when using a mobile and pressing the back button I just exit MASS and get to a "strange nowhere". When I am in MASS and for example is in the radio og albums view, the expected behaviour when pressing the back key on the mobile would be to get to the previous screen the same way as the arrow. Agree?

**Comment by disconn3ct** [comment]
The only thing that is less helpful than the on-screen back button (with it's repeat-screens, "strange nowhere"s and so forth) is the system back button which could land you anywhere, including other applications.

Why can't MA just render in the same framework as everything else? Leave the sidebar alone, don't try to force breadcrumbs. How often do you *really* intend to "walk" around the interface? There is no interactivity there, and no real information. Leaving it open or continuing to interact after activating it seems like an uncommon use pattern.

Trivial example: I Just opened the mobile HA app, clicked "Music Assistant", then "Radio" and selected a radio for the defaulted target. It started playing, so I pushed Back (because there is no other navigation exposed) and it hangs at the HA logo with "Loading data". Eventually the spinner goes away and I am left with the black loading screen. I have to force close the app to get it back. (It could be worse - try doing that interaction via chromecast, with no navigation.)

**Reply by corvy** [reply]
This is exactly the behaviour I have as-well. And that is not very intuitive :)

**Comment by OzGav** [comment]
This is outdated now with MA v2
